cql3/functions/aggregate_fcts.cc

```diff
@@ -91,13 +93,38 @@ class impl_user_aggregate : public aggregate_function::aggregate {
     ::shared_ptr<scalar_function> _finalfunc;
     const bytes_opt _initcond;
     bytes_opt _acc;
+    // auxiliary variables for reusing instances of wasm UDFs for the scalar function
+    std::optional<wasmtime::Store> _store;
+    std::optional<wasmtime::Instance> _instance;
+    std::optional<wasmtime::Func> _func;
```

Comment:
Having a cache at the wasm level means we don't need to repeat the code for UDF/UDA.